### PR TITLE
Address concerns about not following best practices

### LINKED ISSUE #20903: The "Create a test" section should follow the "best practices" article
The "Create a test" section should follow the guidelines specified in [Unit testing best practices](https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices), in particular the naming convention and the practice of avoiding a Setup (constructor in the case of xUnit) method.
---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: e39c2f1b-f41a-bdf9-2646-09379941453d
* Version Independent ID: 422811cb-6287-03af-ba8d-1daebee9744d
* Content: [Unit testing C# code in .NET Core using dotnet test and xUnit - .NET Core](https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-with-dotnet-test)
* Content Source: [docs/core/testing/unit-testing-with-dotnet-test.md](https://github.com/dotnet/docs/blob/master/docs/core/testing/unit-testing-with-dotnet-test.md)
* Product: **dotnet-core**
* GitHub Login: @ardalis
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

Address concerns about not following best practices

Fixes #20903

### CODE REVIEW

File: docs/core/testing/unit-testing-with-dotnet-test.md [hunk 1]
```diff
@@ -136,17 +136,11 @@ namespace Prime.UnitTests.Services
...
-        {
-            _primeService = new PrimeService();
-        }
-
         [Fact]
         public void IsPrime_InputIs1_ReturnFalse()
         {
-            var result = _primeService.IsPrime(1);
+            var primeService = new PrimeService();
+            var result = primeService.IsPrime(1);
```
Comment (tdykstra) [inline docs/core/testing/unit-testing-with-dotnet-test.md:143]:
```suggestion
            bool result = primeService.IsPrime(1);
```

File: docs/core/testing/unit-testing-with-dotnet-test.md [hunk 2]
```diff
@@ -176,7 +170,8 @@ Run `dotnet test`. The test passes.
 Add prime number tests for 0 and -1. You could copy the preceding test and change the following code to use 0 and -1:
 
 ```csharp
-var result = _primeService.IsPrime(1);
+var primeService = new PrimeService();
+var result = primeService.IsPrime(1);
```
Comment (tdykstra) [inline docs/core/testing/unit-testing-with-dotnet-test.md:174]:
```suggestion
bool result = primeService.IsPrime(1);
```

File: docs/core/testing/unit-testing-with-dotnet-test.md [hunk 3]
```diff
@@ -192,7 +187,8 @@ Rather than creating new tests, apply the preceding xUnit attributes to create a
 [Fact]
 public void IsPrime_InputIs1_ReturnFalse()
 {
-    var result = _primeService.IsPrime(1);
+    var primeService = new PrimeService();
+    var result = primeService.IsPrime(1);
```
Comment (tdykstra) [inline docs/core/testing/unit-testing-with-dotnet-test.md:191]:
```suggestion
    bool result = primeService.IsPrime(1);
```

File: docs/core/testing/unit-testing-with-mstest.md [hunk 4]
```diff
@@ -100,17 +100,11 @@ namespace Prime.UnitTests.Services
...
-        {
-            _primeService = new PrimeService();
-        }
-
         [TestMethod]
         public void IsPrime_InputIs1_ReturnFalse()
         {
-            var result = _primeService.IsPrime(1);
+            var primeService = new PrimeService();
+            var result = primeService.IsPrime(1);
```
Comment (tdykstra) [inline docs/core/testing/unit-testing-with-mstest.md:107]:
```suggestion
            bool result = primeService.IsPrime(1);
```

**Review by tdykstra — APPROVED** [review]
Preexisting issue -- suggestion to not use var where the right side doesn't make the inferred type obvious. This is an edge case since you could guess that a name like "IsPrime" would return bool.

**Comment by IEvangelist** [comment]
> Preexisting issue -- suggestion to not use var where the right side doesn't make the inferred type obvious. This is an edge case since you could guess that a name like "IsPrime" would return bool.

You caught me, yeah - I've always hated being explicit... but I'll do it since I was called out on it. :)

**Pull request merged**